examples/ASDKgram/Sample/PhotoTableViewCell.m

```diff
@@ -0,0 +1,248 @@
+  UILabel      *_photoLikesLabel;
+  UILabel      *_photoDescriptionLabel;
+}
+
+#pragma mark - Class Methods
+
++ (CGFloat)heightForPhotoModel:(PhotoModel *)photo withWidth:(CGFloat)width;
+{
+  CGFloat photoHeight = width;
+  
+  UIFont *font        = [UIFont systemFontOfSize:FONT_SIZE];
+  CGFloat likesHeight = roundf([font lineHeight]);
+  
+  NSAttributedString *descriptionAttrString = [photo descriptionAttributedStringWithFontSize:FONT_SIZE];
+  CGFloat availableWidth                    = (width - HORIZONTAL_BUFFER * 2);
+  CGFloat descriptionHeight                 = [descriptionAttrString boundingRectWithSize:CGSizeMake(availableWidth, CGFLOAT_MAX)
+                                                                                  options:NSStringDrawingUsesLineFragmentOrigin
+                                                                                  context:nil].size.height;
+  
+  CGFloat commentViewHeight = [CommentView heightForCommentFeedModel:photo.commentFeed withWidth:availableWidth];
+  
+  return HEADER_HEIGHT + photoHeight + likesHeight + descriptionHeight + commentViewHeight + (4 * VERTICAL_BUFFER);
+}
+
+#pragma mark - Lifecycle
+
+- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
+{
+  self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
+  
+  if (self) {
+    
+    _photoCommentsView                   = [[CommentView alloc] init];
+
+    _userAvatarImageView                 = [[UIImageView alloc] init];
+    _photoImageView                      = [[UIImageView alloc] init];
+
+    _userNameLabel                       = [[UILabel alloc] init];
+    _photoLocationLabel                  = [[UILabel alloc] init];
+    _photoTimeIntervalSincePostLabel     = [[UILabel alloc] init];
+    _photoLikesLabel                     = [[UILabel alloc] init];
+    _photoDescriptionLabel               = [[UILabel alloc] init];
+    _photoDescriptionLabel.numberOfLines = 3;
+
+    [self addSubview:_photoCommentsView];
+    [self addSubview:_userAvatarImageView];
+    [self addSubview:_photoImageView];
+    [self addSubview:_userNameLabel];
+    [self addSubview:_photoLocationLabel];
+    [self addSubview:_photoTimeIntervalSincePostLabel];
+    [self addSubview:_photoLikesLabel];
+    [self addSubview:_photoDescriptionLabel];
+    
+#if DEBUG_PHOTOCELL_LAYOUT
+    _userAvatarImageView.backgroundColor              = [UIColor greenColor];
+    _userNameLabel.backgroundColor                    = [UIColor greenColor];
+    _photoLocationLabel.backgroundColor               = [UIColor greenColor];
+    _photoTimeIntervalSincePostLabel.backgroundColor  = [UIColor greenColor];
+    _photoCommentsView.backgroundColor                = [UIColor greenColor];
+    _photoDescriptionLabel.backgroundColor            = [UIColor greenColor];
+    _photoLikesLabel.backgroundColor                  = [UIColor greenColor];
+#endif
+  }
+  
+  return self;
+}
+
+- (void)layoutSubviews
```

Comment:
Because the comparison implementation uses layout specifications, which are at least as flexible as Autolayout, we should use Autolayout in the UIKit comparison.

This code is fairly clean and worth including however to extend the example by providing an alternative mechanism, For developers who prefer the manual layout style.  Maybe add a #define USE_UIKIT_AUTOLAYOUT 1, and you can compile out the method by default and then a developer testing the app can easily switch it if they are reading the code and interested